python_dwd/parsing_data/parse_data_from_files.py

```diff
@@ -57,12 +60,25 @@ def _parse_dwd_data(filename_and_file: Tuple[str, BytesIO]) -> pd.DataFrame:
             dtype="str"  # dtypes are mapped manually to ensure expected dtypes
         )
     except pd.errors.ParserError:
-        print(f"The file representing {filename} could not be parsed and is skipped.")
+        log.warning(f"The file representing {filename} could not be parsed and is skipped.")
         return pd.DataFrame()
     except ValueError:
-        print(f"The file representing {filename} is None and is skipped.")
+        log.warning(f"The file representing {filename} is None and is skipped.")
         return pd.DataFrame()
 
-    data = data.rename(columns=str.upper).rename(GERMAN_TO_ENGLISH_COLUMNS_MAPPING)
+    # Column names contain spaces, so strip them away.
+    data.rename(columns=str.strip, inplace=True)
+
+    # Make column names uppercase.
+    data.rename(columns=str.upper, inplace=True)
+
+    # End of record (EOR) has no value, so drop it right away.
+    data.drop(columns='EOR', errors='ignore', inplace=True)
 
-    return data.astype(create_stationdata_dtype_mapping(data.columns))
+    # Assign meaningful column names.
+    data.rename(columns=GERMAN_TO_ENGLISH_COLUMNS_MAPPING, inplace=True)
```

Comment:
> I think we shouldn't use the inplace method as of its controversy.

Thanks for outlining that, I've removed the "inplace" feature.

> ... rather overwrite the DataFrame and use the chaining possibilities to reduce the amount of code.

However, I would like to preserve the verbose style in favor of the comments instead of using chaining here. If you still like to compress the code, please let me know.

Reply:
That totally works for me.